### ISSUE TITLE
github workspace setting not working

### ISSUE THREAD

**Issue by yeya24** [thanos-io/thanos#6813]
<img width="931" alt="image" src="https://github.com/thanos-io/thanos/assets/25150124/2b8259e3-89f8-4c5a-9710-a9bd31d22d67">

This is not a bug for Thanos itself. But maybe related to the settings we have here https://github.com/thanos-io/thanos/tree/main/.devcontainer.

Can be reproduced by creating a new code space on main.

**Comment by yeya24** [comment]
Creation log:

```

=================================================================================
2023-10-19 21:46:59.771Z: Configuration starting...
2023-10-19 21:46:59.827Z: Cloning...
2023-10-19 21:46:59.872Z: $ git -C "/var/lib/docker/codespacemount/workspace" clone --branch "main" --depth 1 https://github.com/thanos-io/thanos "/var/lib/docker/codespacemount/workspace/thanos"
2023-10-19 21:46:59.973Z: Cloning into '/var/lib/docker/codespacemount/workspace/thanos'...
2023-10-19 21:47:03.207Z: git process exited with exit code 0
2023-10-19 21:47:03.249Z: $ git -C "/var/lib/docker/codespacemount/workspace/thanos" config --local remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
2023-10-19 21:47:03.266Z: git process exited with exit code 0

=================================================================================
2023-10-19 21:47:04.072Z: Creating container...
2023-10-19 21:47:04.110Z: $ devcontainer up --id-label Type=codespaces --workspace-folder /var/lib/docker/codespacemount/workspace/thanos --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --log-level trace --log-format json --update-remote-user-uid-default never --mount-workspace-git-root false --omit-config-remote-env-from-metadata --skip-non-blocking-commands --skip-post-create --config "/var/lib/docker/codespacemount/workspace/thanos/.devcontainer/devcontainer.json" --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --container-session-data-folder /workspaces/.codespaces/.persistedshare/devcontainers-cli/cache --secrets-file /root/.codespaces/shared/user-secrets-envs.json
2023-10-19 21:47:04.341Z: @devcontainers/cli 0.52.1. Node.js v18.17.1. linux 6.2.0-1014-azure x64.
2023-10-19 21:47:04.791Z: {"outcome":"error","message":"Command failed: docker inspect --type image mcr.microsoft.com/devcontainers/go:0-1.21","description":"An error occurred setting up the container."}
2023-10-19 21:47:04.796Z: Error: Command failed: docker inspect --type image mcr.microsoft.com/devcontainers/go:0-1.21
2023-10-19 21:47:04.799Z:     at EAA (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:462:1260)
2023-10-19 21:47:04.802Z:     at kK (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:462:1002)
2023-10-19 21:47:04.807Z:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2023-10-19 21:47:04.810Z:     at async NAA (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:479:3660)
2023-10-19 21:47:04.822Z:     at async vC (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:479:4775)
2023-10-19 21:47:04.827Z:     at async etA (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:613:10941)
2023-10-19 21:47:04.833Z:     at async AtA (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:613:10682)
2023-10-19 21:47:04.840Z: devcontainer process exited with exit code 1

====================================== ERROR ====================================
2023-10-19 21:47:04.845Z: Failed to create container.
=================================================================================
2023-10-19 21:47:04.848Z: Error: Command failed: docker inspect --type image mcr.microsoft.com/devcontainers/go:0-1.21
2023-10-19 21:47:04.860Z: Error code: 1302 (UnifiedContainersErrorFatalCreatingContainer)

====================================== ERROR ====================================
2023-10-19 21:47:04.872Z: Container creation failed.
=================================================================================
```

**Comment by yeya24** [comment]
Looks like this is the issue https://github.com/thanos-io/thanos/pull/6615/files#diff-13bd9d7a30bf46656bc81f1ad5b908a627f9247be3f7d76df862b0578b534fc6R2

The image tag for 1.21 seems wrong. Available image for 1.21 is listed here https://github.com/devcontainers/images/tree/main/src/go

**Comment by squat** [comment]
Indeed, it looks like rather than `0-1.21` it should be `1.21`
